lms/djangoapps/certificates/views/webview.py

```diff
@@ -334,17 +336,27 @@ def render_html_view(request, user_id, course_id):
             raise CourseDoesNotExist
 
         # Attempt to load the user's generated certificate data
-        user_certificate = GeneratedCertificate.objects.get(
-            user=user,
-            course_id=course_key
-        )
+        if preview_mode:
+            user_certificate = GeneratedCertificate.objects.get(
+                user=user,
+                course_id=course_key,
+                mode=preview_mode
+            )
+        else:
+            user_certificate = GeneratedCertificate.objects.get(
+                user=user,
+                course_id=course_key
+            )
 
     # If there's no generated certificate data for this user, we need to see if we're in 'preview' mode...
     # If we are, we'll need to create a mock version of the user_certificate container for previewing
     except GeneratedCertificate.DoesNotExist:
-        if request.GET.get('preview', None):
+        if preview_mode and (
+            has_access(request.user, 'instructor', course)
+            or has_access(request.user, 'staff', course)
+        ):
```

Comment:
I know, In past It was in my mind to restrict unauthorized users to preview certificate, if someone add the ?preview='xyz' in URL. 
good to know, we are handling this scenario now :)